src/main/java/org/radarcns/management/web/rest/SubjectResource.java

```diff
@@ -377,12 +427,13 @@
         Long sourceTypeId = sourceDto.getSourceTypeId();
         if (sourceTypeId == null) {
             // check if combination (producer, model, version) is present
-            final String msg = "error.sourceNotFound";
             try {
-                String producer = Objects.requireNonNull(sourceDto.getSourceTypeProducer(), msg);
-                String model = Objects.requireNonNull(sourceDto.getSourceTypeModel(), msg);
+                String producer = Objects.requireNonNull(sourceDto.getSourceTypeProducer(),
+                        ErrorConstants.ERR_SOURCE_NOT_FOUND);
+                String model = Objects.requireNonNull(sourceDto.getSourceTypeModel(),
```

Comment:
should it be `SOURCE_TYPE_NOT_FOUND`?

Reply:
Actually these checks should throw a validation error if any of these are null, because that would be a bad request from the client. Will change.